src/sentry/lang/native/utils.py

```diff
@@ -72,6 +78,32 @@ def find_all_stacktraces(data):
     return rv
 
 
+def get_sdk_from_event(event):
+    sdk_info = (event.get('debug_meta') or {}).get('sdk_info')
+    if sdk_info:
+        return sdk_info
+    os = (event.get('contexts') or {}).get('os')
+    if os and os.get('type') == 'os':
+        return get_sdk_from_os(os)
+
+
+def get_sdk_from_os(data):
+    if 'name' not in data or 'version' not in data:
+        return
+    dsym_type = KNOWN_DSYM_TYPES.get(data['name'])
+    if dsym_type is None:
+        return
+    system_version = tuple(int(x) for x in (
```

Comment:
For sake of being nitpicky, it'd be marginally faster to do:

```python
map(int, ...)
```

rather than `tuple(int(x) for x in ...)`

But pretty insignificant. I won't block for this.